src/unstract/sdk/file_storage/fs_impl.py

```diff
@@ -308,25 +311,51 @@ def get_hash_from_file(self, path: str) -> str:
             h = sha256()
             b = bytearray(128 * 1024)
             mv = memoryview(b)
-            with self._open_with_no_buffering(path=path) as f:
+            with self.fs.open(path) as f:
                 while n := f.readinto(mv):
                     h.update(mv[:n])
             return str(h.hexdigest())
         except Exception as e:
             raise FileOperationError(str(e)) from e
 
-    def json_dump(self, path, mode, encoding, data, **kwargs):
+    def json_dump(
+        self,
+        path: str,
+        data: Union[str, bytes],
```

Comment:
I think it would be clearer if the data parameter is described as JSON data. This makes it clear that the method is for dumping JSON into a file. And we can avoid to convert the json to str everywhere we using this method.

Reply:
Made this data a dict now.